### ISSUE TITLE
Communication about change command file format

### ISSUE THREAD

**Issue by bestony** [tldr-pages/tldr#2835]
### Who are we ?

We are the maintainer of Linux.CN , We Provide Chinese Linux Information and Article for people who couldn't read the English article.

### What we Do?

We Build a WeChat MiniProgram Client For tldr. 

This client is published at [Feb 12,2019](https://linux.cn/article-10528-1.html) ,

We use this client to make more people help translate tldr project into Chinese.

![](https://i.loli.net/2019/03/17/5c8e5d6628508.png)

After one month  of operation, Now We Have 409 Fully translated Items, and 164 half translated items, with more than 2784 command examples translated.

![](https://i.loli.net/2019/03/17/5c8e5d805485e.png)

we provide a command request function to make people notify me which command is missing.now ,we have more than 700 commands need to add.

![](https://i.loli.net/2019/03/17/5c8e6275830bd.png)

Now, More than 16K people use our MiniProgram Client.

![](https://i.loli.net/2019/03/17/5c8e5dd2adf73.png)

### What We Want about this issue 

As a open source community, we want export our Database into this project. (Upload Our Data to upstream).

### The problem We Meet

Our MiniProgram Client is not base on Git, We Track it in our database, Now, We will upload data to upstream, and we want to make all contributors show for each command, then , their contributions will track by git.

But the problem is **TLDR Format not support show contributor information**. If we want to show some contributor information , we must add some meta information in files.

### What We think should can help slove this problem

At here, we think there has two different solutions can help with this problem.

#### Solution 1: add some meta information into files

the new format file like this(I add a footnote for contributor)

```markdown
# as [^contributor]

> Portable GNU assembler. 
> Primarily intended to assemble output from `gcc` to be used by `ld`.

- Assemble a file, writing the output to a.out:

`as {{file.s}}`

- Assemble the output to a given file:

`as {{file.s}} -o {{out.o}}`

- Generate output faster by skipping whitespace and comment preprocessing. (Should only be used for trusted compilers):

`as -f {{file.s}}`

- Include a given path to the list of directories to search for files specified in .include directives:

`as -I {{path/to/directory}} {{file.s}}`

[^contributor]: Bestony, John, Tom
```

#### Solution 2: We Create a new Fork for TLDR

If the file format cann't change, i think we should make a fork for tldr, and contribute it by ourselves. 

**If we contribute the fork, We want to the official client can support change data source** ,then ,people who live in China can change source to view the chinese version.

and other country people can make a fork for their language.

**Use your wechat scanner scan this code,  you can view our client**

![gh_133ae827415c_258](https://user-images.githubusercontent.com/13283837/54499213-3a896080-494a-11e9-9a6c-fe2308f94e8c.jpg)

**Comment by bestony** [comment]
@starccy 
@agnivade

**Comment by agnivade** [comment]
The official node client does indeed support changing the data source. So solution 2 is always an option.

That said, we would really appreciate if you can send PRs for the translated pages. Yes, we have always tracked contributions from the git commit history. Since you haven't used git for the translated pages, it is indeed an issue. I have a suggestion which you can think about - similar to MAINTAINERS.md, we can have a CONTRIBUTORS.md where we can mention the name/email of the person who has contributed the page. That is a possibiblity. Let me know your thoughts.

**Comment by sbrl** [comment]
Hey, wow! That's a _lot_ of translations.

Git lets you change the email address and username when committing. If your users have an email  address and username on GitHub, could you write a program that generates git commits with those credentials? Then the commits will be credited to the correct accounts.

It's up to the implementing client, but it doesn't sound like a huge deal to write a client that shows the users that have contributed to a page by inspecting the git history. It all depends on the way a client is implemented. To this end, we might need to alter the index.json files we auto-generate to include this information? See #2829.

**Comment by agnivade** [comment]
I think OP's concern was just to show user's contribution. If we can attribute the git commit to the user's email address, that will also work.

**Comment by bestony** [comment]
> Hey, wow! That's a _lot_ of translations. Thanks!
> 
> Git lets you change the email address and username when committing. If your users have an email address and username on GitHub, could you write a program that generates git commits with those credentials? Then the commits will be credited to the correct accounts.
> 
> It's up to the implementing client, but it doesn't sound like a huge deal to write a client that shows the users that have contributed to a page by inspecting the git history. It all depends on the way a client is implemented. To this end, we might need to alter the index.json files we auto-generate to include this information? See #2829.

The Question is WeChat only provide a Nickname, No Email. Maybe i should add some settings into our client to make people can input their emaill and github name.

**Comment by bestony** [comment]
Another Question is as we can see , we have more than 700+ command request from our client. We will add some item for them ,but we maybe direct add chinese item, not english.

Is this ok?

**Comment by bestony** [comment]
i will open an issue for export data with `contributors.md` and make a pull request to discuss

**Comment by agnivade** [comment]
>  we have more than 700+ command request from our client. We will add some item for them ,but we maybe direct add chinese item, not english.

Well, the translated pages are meant to be in Chinese right ? So yes it should be in Chinese. Unless I am misunderstanding something ?

**Comment by wxy** [comment]
> Another Question is as we can see , we have more than 700+ command request from our client. We will add some item for them ,but we maybe direct add chinese item, not english.
> 
> Is this ok?

No, if the requested items are pushed, we will submit the English version at the same time.  @bestony

**Comment by wxy** [comment]
> > we have more than 700+ command request from our client. We will add some item for them ,but we maybe direct add chinese item, not english.
> 
> Well, the translated pages are meant to be in Chinese right ? So yes it should be in Chinese. Unless I am misunderstanding something ?

No. Now we have 409 Fully translated Items, and 164 half translated items, with more than 2784 command examples translated. 700+ entries for new command requests are not in the above range.

**Comment by agnivade** [comment]
Oh you are talking about new pages. Sure, any language is fine. But please do not send all 409 pages in a single shot. For translations, it is fine. But for english pages, they will have to undergo a review cycle, which takes time. For english pages, I would suggest to send 5 pages at a time.

**Comment by sbrl** [comment]
I'd also suggest sending the English pages first, and not submitting 2 different languages in the same PR. In other words, keep English pages in 1 set of PRs, and translations in another (if multiple other languages are involved, perhaps 1 set of PRs per language?)

**Comment by bestony** [comment]
i will export all our data as soon as possible. then, i will paste a link here.

**Comment by zero-MK** [comment]
We have now completed the translation
![S00706-18581834](https://user-images.githubusercontent.com/36980619/86586673-db29e680-bfba-11ea-97f3-a449e95cdb7b.jpg)

**Comment by zero-MK** [comment]
We have now completed the translation
![S00706-18581834](https://user-images.githubusercontent.com/36980619/86586926-525f7a80-bfbb-11ea-82d6-e56f6c27bc34.jpg)
@agnivade 
@sbrl

**Comment by sbrl** [comment]
Hey, cool! So the next steps here are to import the translations into the tldr-pages repo then?

In that case, perhaps we could go in this order:

1. Submit a PR for the chinese translations of pages that already exist in tldr-pages
2. Submit a PRs for the english versions of new pages (5 pages at a time)
3. Submit chinese translations for the new pages

Does this sound ok?

**Comment by waldyrious** [comment]
I'm not sure the contributors' attribution issue is settled. The relevant quotes from the thread above:

> `@bestony`: Our MiniProgram Client is not base on Git, We Track it in our database, Now, We will upload data to upstream, and we want to make all contributors show for each command, then , their contributions will track by git.

> `@sbrl`: Git lets you change the email address and username when committing. If your users have an email address and username on GitHub, could you write a program that generates git commits with those credentials? Then the commits will be credited to the correct accounts.

> `@agnivade`: I think OP's concern was just to show user's contribution. If we can attribute the git commit to the user's email address, that will also work.

> `@bestony`: The Question is WeChat only provide a Nickname, No Email. Maybe i should add some settings into our client to make people can input their emaill and github name.

Unless I missed something, there wasn't further discussion on this topic. I just wanted to add that, if we decide to go with the git approach, it's not necessary that we use real emails for the commits. [Here](https://stackoverflow.com/a/7373004/266309), for example, it is suggested that the git author email can be empty (like `John Smith <>`). This means that using the WeChat nickname as the author name would be sufficient to provide attribution.

That said, @agnivade's proposal to use CONTRIBUTORS.md might indeed be the easiest approach:

> `@agnivade`: I have a suggestion which you can think about - similar to MAINTAINERS.md, we can have a CONTRIBUTORS.md where we can mention the name/email of the person who has contributed the page. That is a possibiblity. Let me know your thoughts.

> `@bestony`: i will open an issue for export data with `contributors.md` and make a pull request to discuss

Was there any follow-up on this? Was any decision made?

**Comment by agnivade** [comment]
Yeah, updating CONTRIBUTORS.md sounds good. This was the real problem. Adding the pages was never an issue.

**Comment by Managor** [comment]
@bestony Do you still have interest in this?

**Comment by Managor** [comment]
Since there has been no further interest in this issue, I will close this.
